cpp/arcticdb/version/version_store_api.cpp

```diff
@@ -473,15 +474,28 @@ void PythonVersionStore::snapshot(
         std::set_difference(all_symbols.begin(), all_symbols.end(), skip_symbols_set.begin(),
                 skip_symbols_set.end(), std::back_inserter(filtered_symbols));
 
-        if (filtered_symbols.empty()) {
-            log::version().warn("No valid symbols in the library, skipping creation for snapshot: {}", snap_name);
-            return;
-        }
+        missing_data::check<ErrorCode::E_NO_SUCH_VERSION>(
+            !filtered_symbols.empty(),
+            "No valid symbols in the library, skipping creation for snapshot: {}", snap_name
+        );
 
         auto sym_index_map = batch_get_latest_version(store(), version_map(), filtered_symbols, false);
         index_keys = utils::values(*sym_index_map);
     } else {
-        auto sym_index_map = batch_get_specific_version(store(), version_map(), versions);
+        auto sym_index_map = batch_get_specific_version(store(), version_map(), versions, BatchGetVersionOption::LIVE_AND_TOMBSTONED_VER_REF_IN_OTHER_SNAPSHOT);
+        if (allow_partial_snapshot) {
+            missing_data::check<ErrorCode::E_NO_SUCH_VERSION>(
+                !sym_index_map->empty(),
+                "No symbol-version pair specified in versions cannot be found in the library, skipping creation for snapshot: {}",
```

Comment:
Double negative. I would go for:
`None of the symbol-version pairs specified in versions exist, skipping creation for snapshot: {}`